### PR TITLE
issue-1440: button states + logging use enums

### PR DESCRIPTION
**Pull request by oliverfoster**
Converted core enums to use ENUM behaviour.

fixes: https://github.com/adaptlearning/adapt_framework/issues/1440

* Added button state enum
* Switch questionView questionModel and buttonView to use button state enum
* Added logging level enum
* Switched logging.js to use logging level enum
* Flattened logging.js logic to make a bit less nested and a bit clearer to read (https://github.com/adaptlearning/adapt_framework/blob/5ca8849a6711e54a7171e3d3e050f8e89c306b0f/src/core/js/logging.js)
* Fixed/upgraded some require statements
* Fixed some indentation on switch statements
* Upgraded enum.js to deal with upper and lower case name

### CODE REVIEW

File: src/core/js/enums/loggingLevelEnum.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+define(function() {
+
+    // Used to determine if log call should be printed based on log level
+    var LOGGING_LEVEL = ENUM([
```
Comment (moloko) [inline src/core/js/enums/loggingLevelEnum.js]: `LOG_LEVEL` would get the same meaning across with 4 less chars to type...

File: src/core/js/enums/questionButtonStateEnum.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+define(function() {
+
+    var QUESTION_BUTTON_STATE = ENUM([
```
Comment (moloko) [inline src/core/js/enums/questionButtonStateEnum.js]: `BUTTON_STATE` would not only be shorter but would also allow for inclusion of non-question button states in the future

**Comment by oliverfoster** [comment]
done @brian-learningpool

**Comment by brian-learningpool** [comment]
Looks good.

File: src/core/js/views/buttonsView.js [hunk 3]
```diff
@@ -91,12 +95,14 @@ define(function() {
...
 
             } else {
-                // Backwords compatibility with v1.x
-                var ariaLabel = this.model.get('_buttons')["_" + changedAttribute].ariaLabel;
-                var buttonText = this.model.get('_buttons')["_" + changedAttribute].buttonText;
 
-                switch (changedAttribute) {
-                    case "showCorrectAnswer": case "hideCorrectAnswer":
+                // Backwords compatibility with v1.x               
+                var ariaLabel = this.model.get('_buttons')["_" + buttonState.asString].ariaLabel;
```
Comment (tomgreenfield) [inline src/core/js/views/buttonsView.js]: `buttonState.asString` won't suffice – needs underscores removing and converting to camel case.

File: src/core/js/views/buttonsView.js [hunk 4]
```diff
@@ -96,12 +108,14 @@ define([
 
             } else {
 
-                // Backwords compatibility with v1.x               
-                var ariaLabel = this.model.get('_buttons')["_" + buttonState.asString].ariaLabel;
-                var buttonText = this.model.get('_buttons')["_" + buttonState.asString].buttonText;
+                // Backwords compatibility with v1.x
+                var propertyName = textPropertyName[buttonState.asString];
+
+                var ariaLabel = this.model.get('_buttons')["_" + propertyName] .ariaLabel;
```
Comment (tomgreenfield) [inline src/core/js/views/buttonsView.js]: Extra space on this line.

**Pull request merged**